Controllers/controller.py

```diff
@@ -106,3 +111,17 @@ def open_file_dialog(self):
             url = file_dialog.selectedUrls()[0]
             self._media_player.setSource(url)
             self._media_player.play()
+
+    @Slot()
+    def position_changed(self):
+        position = self._media_player.position()
+        self._window.media_panel.slider.setValue(position)
+
+    @Slot()
+    def duration_changed(self):
```

Comment:
Add a docstring for the duration_changed function. Use the same style as the other docstrings (look at other function docstrings).